sql/core/src/test/scala/org/apache/spark/sql/execution/SameResultSuite.scala

```diff
@@ -46,4 +48,10 @@ class SameResultSuite extends QueryTest with SharedSQLContext {
     df.queryExecution.sparkPlan.find(_.isInstanceOf[FileSourceScanExec]).get
       .asInstanceOf[FileSourceScanExec]
   }
+
+  test("SPARK-20725: partial aggregate should behave correctly for sameResult") {
+    val df1 = spark.range(10).agg(sum($"id"))
+    val df2 = spark.range(10).agg(sum($"id"))
```

Comment:
Good catch! The reason is, `HashAggregateExec.requiredChildDistributionExpressions` is a `Option[Seq[Expression]]`, which is not treated as expressions of `HashAggregateExec`, and thus not touched by `QueryPlan.mapExpressions`.

I have fixed it in `QueryPlan`